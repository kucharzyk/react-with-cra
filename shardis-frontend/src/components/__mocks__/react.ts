import * as React from 'react';

// Temporary hack to suppress error
// https://github.com/facebookincubator/create-react-app/issues/3199
window.requestAnimationFrame = function (callback) {
  setTimeout(callback, 0);
  return 0;
};

module.exports = React;
