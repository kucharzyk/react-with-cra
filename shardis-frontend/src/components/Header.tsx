import * as React from 'react';
import { StatelessComponent } from 'react';

const logo = require('./logo.svg');

const Header: StatelessComponent = () => (
  <div className="App-header">
    <img src={logo} className="App-logo" alt="logo"/>
    <h2>Welcome to React</h2>
  </div>
);

export default Header;
