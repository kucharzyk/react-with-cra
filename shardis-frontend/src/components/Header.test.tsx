import * as React from 'react';
import * as ReactDOM from 'react-dom';

import Header from './Header';
import {create} from 'react-test-renderer';

describe('Header', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header/>, div);
  });

  it('renders correctly', () => {
    const tree = create(<Header/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
