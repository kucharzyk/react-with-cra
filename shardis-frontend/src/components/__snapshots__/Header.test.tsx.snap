// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="App-header"
>
  <img
    alt="logo"
    className="App-logo"
    src="logo.svg"
  />
  <h2>
    Welcome to React
  </h2>
</div>
`;
